explorer/src/Pos/Explorer/TestUtil.hs

```diff
@@ -0,0 +1,240 @@
+
+-- I used the build function since I suspect that it's safe (even in tests).
+instance HasConfiguration => Prelude.Show SlogUndo where
+    show = show . build
+
+instance Prelude.Show DlgUndo where
+    show = show . build
+
+instance HasConfiguration => Prelude.Show Undo where
+    show = show . build
+
+instance Arbitrary SlogUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary DlgUndo where
+    arbitrary = genericArbitrary
+
+instance HasConfiguration => Arbitrary Undo where
+    arbitrary = genericArbitrary
+
+----------------------------------------------------------------
+-- Utility
+-- TODO(ks): Extract this in some common PureBlockTest module in src/?
+----------------------------------------------------------------
+
+basicBlockGenericUnsafe
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => BlockHeader
+    -> SecretKey
+    -> SlotId
+    -> Block
+basicBlockGenericUnsafe prevHeader sk slotId = case (basicBlock prevHeader sk slotId) of
+    Left e      -> error e
+    Right block -> Right block
+
+basicBlock
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => BlockHeader
+    -> SecretKey
+    -> SlotId
+    -> Either Text MainBlock
+basicBlock prevHeader sk slotId =
+    producePureBlock infLimit prevHeader [] Nothing slotId def (defGTP slotId) def sk
+
+emptyBlk
+    :: (HasConfiguration, HasUpdateConfiguration, Testable p)
+    => (Either Text MainBlock -> p)
+    -> Property
+emptyBlk testableBlock =
+    forAll arbitrary $ \(sk, prevHeader, slotId) ->
+    testableBlock
+        $ producePureBlock infLimit prevHeader [] Nothing slotId def (defGTP slotId) def sk
+
+defGTP :: HasConfiguration => SlotId -> SscPayload
+defGTP sId = defaultSscPayload $ siSlot sId
+
+infLimit :: Byte
+infLimit = convertUnit @Gigabyte @Byte 1
+
+producePureBlock
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => Byte
+    -> BlockHeader
+    -> [TxAux]
+    -> ProxySKBlockInfo
+    -> SlotId
+    -> DlgPayload
+    -> SscPayload
+    -> UpdatePayload
+    -> SecretKey
+    -> Either Text MainBlock
+producePureBlock limit prev txs psk slot dlgPay sscPay usPay sk =
+    createMainBlockPure limit prev psk slot sk $
+    RawPayload txs sscPay dlgPay usPay
+
+leftToCounter :: (ToString s, Testable p) => Either s a -> (a -> p) -> Property
+leftToCounter x c = either (\t -> counterexample (toString t) False) (property . c) x
+
+type BlockNumber   = Word
+type SlotsPerEpoch = Word
+type TotalEpochs   = Word
+
+-- | Function that should generate arbitrary blocks that we can use in tests.
+produceBlocksByBlockNumberAndSlots
+    :: forall m. (HasConfiguration, HasUpdateConfiguration, MonadIO m, Monad m)
+    => BlockNumber
+    -> SlotsPerEpoch
+    -> SlotLeaders
+    -> [SecretKey]
+    -> m [Block]
+produceBlocksByBlockNumberAndSlots blockNumber slotsNumber producedSlotLeaders secretKeys = do
+
+    -- This is just plain wrong and we need to check for it.
```

Comment:
Should we create an issue and/or add TODO/FIXME tag ?

Reply:
I'm still working on the PR, but I think I just forgot to remove that line. Thanks for the reminder.